src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -292,9 +293,14 @@ public void runDevAppServer1Command(List<String> jvmArgs, List<String> args)
 
     logCommand(command);
 
-    Map<String, String> environment = Maps.newHashMap();
-    environment.put("JAVA_HOME", javaHomePath.toAbsolutePath().toString());
-    processRunner.setEnvironment(environment);
+    Map<String, String> devServerEnvironment = Maps.newHashMap(environment);
+    if (!devServerEnvironment.isEmpty()) {
+      logger.info("Setting appengine-web.xml configured environment variables: "
```

Comment:
OK now I am being a total PITA but I worry that some other entrypoint might pass environment that's not based off of appengine-web.xml. Perhaps this logging statement belongs closer to CloudSdkAppEngineDevServer1:120?

Feel free to throw darts at an effigy of me.